docs/en/rules/Azure.AKS.AzureCNI.md

```diff
@@ -0,0 +1,31 @@
+---
+severity: Important
+pillar: Operational Excellence
+category: Capacity planning
+resource: Azure Kubernetes Service
+online version: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.AKS.AzureCNI/
+---
+
+# AKS clusters using Azure CNI should use large subnets
+
+## SYNOPSIS
+
+AKS clusters using Azure CNI should use large subnets to reduce IP exhaustion issues.
+
+## DESCRIPTION
+
+In addition to kubenet, AKS clusters support Azure Container Networking Interface (CNI). This enables every pod to be accessed directly from the subnet via an IP address. Each node supports a maximum number of pods, which are reserved as IP addresses. This approach requires more capacity planning ahead of time, and can result to IP address exhaustion or the need to rebuild AKS clusters into a larget subnet as application workloads begin to grow.
+
+## RECOMMENDATION
+
+Consider allocating a large subnet(/23 or bigger) to your AKS cluster.
+
+## NOTES
+
+This rule applies when analyzing resources deployed to Azure.
+
+## LINKS
+
+- [Configure Azure CNI networking in Azure Kubernetes Service (AKS)](https://docs.microsoft.com/azure/aks/configure-azure-cni)
```

Comment:
Add a link to the WAF. I think this would be the closest to the intent of the rule, but if you have a better reference use that.

https://docs.microsoft.com/azure/architecture/framework/scalability/design-scale#plan-for-growth

Possibly if you don't have a better reference for _Operational Excellence_, lets put this rule under _Performance Efficiency_ with the category _Scalabilty_.

Reply:
Thanks @BernieWhite, added that WAF link in. Also changed pillar to Performance Efficiency and Scalability category.